crates/biome_json_analyze/src/assist/source/use_sorted_keys.rs

```diff
@@ -83,6 +105,24 @@ declare_source_rule! {
     }
 }
 
+/// Determines the nesting depth of a JSON value for grouping purposes.
+/// Objects and multi-line arrays are considered nested (depth 1).
+/// Primitives and single-line arrays are considered simple (depth 0).
+fn get_nesting_depth(value: &AnyJsonValue) -> u8 {
+    match value {
+        AnyJsonValue::JsonObjectValue(_) => 1,
+        AnyJsonValue::JsonArrayValue(array) => {
+            // Check if array spans multiple lines by looking for newlines
+            if array.to_string().contains('\n') {
```

Comment:
Shouldn't you use `            if array.syntax().text_trimmed().contains_char('\n') {`?